emoji/src/main/java/com/vanniktech/emoji/EmojiHandler.java

```diff
@@ -2,83 +2,28 @@
 
 import android.content.Context;
 import android.text.Spannable;
-import com.vanniktech.emoji.emoji.Emoji;
+import com.vanniktech.emoji.EmojiManager.EmojiRange;
 import java.util.ArrayList;
 import java.util.List;
 
-import static com.vanniktech.emoji.EmojiHandler.SpanRangeList.SPAN_NOT_FOUND;
-
 final class EmojiHandler {
-  static void addEmojis(final Context context, final Spannable text, final int emojiSize) {
-    final SpanRangeList existingSpanRanges = new SpanRangeList(text);
+  static void replaceWithImages(final Context context, final Spannable text, final int emojiSize) {
     final EmojiManager emojiManager = EmojiManager.getInstance();
-    int index = 0;
-
-    while (index < text.length()) {
-      final int existingSpanEnd = existingSpanRanges.spanEnd(index);
-
-      if (existingSpanEnd == SPAN_NOT_FOUND) {
-        final int nextSpanStart = existingSpanRanges.nextSpanStart(index);
-        final int searchRange = nextSpanStart == SPAN_NOT_FOUND ? text.length() : nextSpanStart;
-        final Emoji found = emojiManager.findEmoji(text.subSequence(index, searchRange));
+    final List<Integer> existingSpans = new ArrayList<>();
```

Comment:
extract `text.getSpans(0, text.length(), EmojiSpan.class)` and then use the size to initialize the List so that it won't increase it's internal limit while adding items to reduce allocations